pkg/util/filtered_cache.go

```diff
@@ -0,0 +1,571 @@
+// Copyright 2020 IBM Corporation
```

Comment:
Can we directly put the IBM source code into our code? You may bring it to open source meeting to discuss?

Reply:
```
I think it depends on the license on the IBM repo you are interested in.  ie is it Apache 2.0? We obviously can't use any closed source IBM stuffs and share it openly.
```
Above is the answer from Mike


@clyang82 @morvencao 
do you think we need to use current way, or fork the repo to our org and use it with our fix?

Reply:
I think we may use it in the current way, since the license is Apache 2 and we keep the copyright information.

Reply:
It's apache v2 license.
I think the current header is correct, we have to keep the original copyright at the top of the file and also what we changed.